src/app/_services/depublication.service.ts

```diff
@@ -149,11 +146,44 @@ export class DepublicationService {
     page: number,
     sort?: SortParameter,
     filter?: string
-  ): Observable<Results<RecordDepublicationInfo>> {
+  ): Observable<DatasetDepublicationInfo> {
     const sortParam = this.parseSortParameter(sort);
     const filterParam = this.parseFilterParameter(filter);
     const url = `${apiSettings.apiHostCore}/depublish/record_ids/${datasetId}?page=${page}${sortParam}${filterParam}`;
-    return this.http.get<Results<RecordDepublicationInfo>>(url).pipe(this.errors.handleRetry());
+    return this.http.get<DatasetDepublicationInfo>(url).pipe(this.errors.handleRetry());
+  }
+
+  /** flattenDatasetDepublicationInfos
+  /*  retrieve publication information
+  /*  @param {function} getResults - the data function
+  /*  @param {number} endPage - the last page to get
+  */
+  flattenDatasetDepublicationInfos(
+    getResults: (page: number) => Observable<DatasetDepublicationInfo>,
+    endPage: number
+  ): Observable<DatasetDepublicationInfo> {
+    const observables: Observable<DatasetDepublicationInfo>[] = [];
+    for (let i = 0; i <= endPage; i++) {
+      observables.push(getResults(i));
+    }
+    return forkJoin(observables).pipe(
+      map((resultList) => {
+        const lastResult = resultList[resultList.length - 1];
+        const allResults = ([] as RecordDepublicationInfo[]).concat(
+          ...resultList.map((item) =>
+            ([] as RecordDepublicationInfo[]).concat(...item.depublicationRecordIds.results)
+          )
+        );
```

Comment:
I think you can simplify this to the following:
```
        const allResults = ([] as RecordDepublicationInfo[]).concat(
          ...resultList.map(item => item.depublicationRecordIds.results)
        );
```
Is that correct?

Reply:
Done